src/codegen/type/type.cpp

```diff
@@ -18,15 +18,20 @@ namespace peloton {
 namespace codegen {
 namespace type {
 
-Type::Type() : Type(peloton::type::TypeId::INVALID, false) {}
-
 Type::Type(peloton::type::TypeId type_id, bool _nullable)
-    : type_id(type_id), nullable(_nullable) {}
+    : type_id(type_id), nullable(_nullable) {
+  aux_info.varlen = 0;
+  aux_info.numeric_info.precision = 0;
+  aux_info.numeric_info.scale = 0;
+}
+
+Type::Type() : Type(peloton::type::TypeId::INVALID, false) {}
 
 Type::Type(const SqlType &sql_type, bool _nullable)
     : Type(sql_type.TypeId(), _nullable) {}
 
 bool Type::operator==(const Type &other) const {
+  // TODO(pmenon): This isn't correct; we need to check all other fields ...
```

Comment:
Shouldn't this be an easy fix?

Reply:
I tried this, but it's a little more involved because of some assumptions we make in the code. We can probably address this in a separate PR.